test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -391,5 +390,34 @@ public class HubConnectionProtocolTests
                 await connection.DisposeAsync().OrTimeout();
             }
         }
+
+        [Fact]
+        public async Task AcceptsPingMessages()
+        {
+            var connection = new TestConnection(TransferMode.Text);
+            var hubConnection = new HubConnection(connection,
+                new JsonHubProtocol(), new LoggerFactory());
+
+            try
+            {
+                // Receive the ping mid-invocation so we can see that the rest of the flow works fine
+
+                await hubConnection.StartAsync().OrTimeout();
+
+                // Ignore negotiate message
+                await connection.ReadSentTextMessageAsync().OrTimeout();
+
+                // Receive a ping
+                await connection.ReceiveJsonMessage(new { type = 6 }).OrTimeout();
+
+                // Send an invocation
+                var invokeTask = hubConnection.InvokeAsync("Foo");
```

Comment:
should this call be awaited?

Reply:
Yes, in fact it's in the wrong order. I'll fix that, not sure how it got messed up like this.

Reply:
Yeah, this test looked like it was not finished...

Reply:
It was finished, and then I did some refactoring and it got unfinished :P